### PR TITLE
add support for source AMI IDs and make source AMI filters variables

### PR DESCRIPTION
**Pull request by dan-hill2802**
Signed-off-by: Daniel Hill <danhill@digital.uc.dwp.gov.uk>

### CODE REVIEW

**Comment by dan-hill2802** [comment]
Don't merge until pipelines are updated

**Comment by mark-r-m** [comment]
@dan-hill2802 is this at odds with what @ClarksonCJ has done? Or is it to allow this to come from another base image?

File: dw-hardened-ami/generic_packer_template.json.j2 [hunk 1]
```diff
@@ -6,14 +6,17 @@
     },
     "builders": [{
       "type": "amazon-ebs",
+      {% if 'source_ami_id' in event and event['source_ami_id']|length > 0 %}
+        "source_ami": "{{ event['source_ami_id']}}"",
+      {% endif %}
```
Comment (ClarksonCJ) [inline dw-hardened-ami/generic_packer_template.json.j2]: The filter should be an else clause if the source ami id is absent should it not?

File: dw-hardened-ami/generic_packer_template.json.j2 [hunk 2]
```diff
@@ -6,14 +6,17 @@
...
       "source_ami_filter": {
         "filters": {
-        "virtualization-type": "hvm",
-        "name": "amzn-ami-hvm-*",
-        "root-device-type": "ebs",
+        "virtualization-type": "{{ event['source_ami_virt_type'] or 'hvm' }}",
+        "name": "{{ event['source_ami_name'] or 'amzn-ami-hvm-*' }}",
+        "root-device-type": "{{ event['source_ami_root_device_type'] or 'ebs' }}",
         "architecture": "x86_64"
```
Comment (ClarksonCJ) [inline dw-hardened-ami/generic_packer_template.json.j2]: Given that we are pinning all versions to AL1, hvm ebs on 64bit architecture, there is no need for the event variables anymore here. The base AL1 has to be compatible with EMR, so deviating from hvm, x86_64 or ebs will break that compatibility. That makes these event vars moot

File: dw-hardened-ami/generic_packer_template.json.j2 [hunk 3]
```diff
@@ -6,14 +6,17 @@
...
         },
-        "owners": ["137112412989"],
+        "owners": ["{{ event['source_ami_owner'] or '137112412989' }}"],
```
Comment (ClarksonCJ) [inline dw-hardened-ami/generic_packer_template.json.j2]: the proper amazon issued AL! AMI is owned by Amazon (aka 137112412989), and as per the above comment, it makes the event variable moot.

File: dw-hardened-ami/generic_packer_template.json.j2 [hunk 4]
```diff
@@ -6,14 +6,17 @@
     },
     "builders": [{
       "type": "amazon-ebs",
+      {% if 'source_ami_id' in event and event['source_ami_id']|length > 0 %}
+        "source_ami": "{{ event['source_ami_id']}}"",
+      {% endif %}
       "source_ami_filter": {
```
Comment (ClarksonCJ) [inline dw-hardened-ami/generic_packer_template.json.j2]: The filter should be an else clause if the source ami id is absent should it not?

**Pull request merged**